spec/ulid_spec.cr

```diff
@@ -57,4 +57,72 @@ describe ULID do
       end
     end
   end
+
+  describe ".valid?" do
+    it "should validate a valid string" do
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3").should be_true
+    end
+
+    it "should be case insensitive" do
+      ULID.valid?("01b3EAF48P97R8MP9WS6MHDTZ3").should be_true
+    end
+
+    it "should detect incorrect length" do
+      ULID.valid?("0").should be_false
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ32").should be_false
+      ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ").should be_false
+      ULID.valid?("!@#$%^&*(").should be_false
+      ULID.valid?("abcde").should be_false
+      ULID.valid?("1234567890").should be_false
+      ULID.valid?("").should be_false
+    end
+
+    it "should detect invalid characters" do
+      ULID.valid?("01!3EAF48P97R8MP9WS8MHDTZ3").should be_false
+      ULID.valid?("01I3EAF48P97R8MP9WS8MHDTZ3").should be_false
+      ULID.valid?("01O3EAF48P97R8MP9WS8MHDTZ3").should be_false
+    end
+  end
+
+  describe ".valid!" do
+    it "should validate a valid string" do
+      ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ3").should be_nil
+    end
+
+    it "should be case insensitive" do
+      ULID.valid!("01b3EAF48P97R8MP9WS6MHDTZ3").should be_nil
+    end
+
+    it "should detect incorrect length" do
+      expect_raises(IncorrectLength) { ULID.valid!("0") }
+      expect_raises(IncorrectLength) { ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ32") }
+      expect_raises(IncorrectLength) { ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ") }
+      expect_raises(IncorrectLength) { ULID.valid!("!@#$%^&*(") }
+      expect_raises(IncorrectLength) { ULID.valid!("abcde") }
+      expect_raises(IncorrectLength) { ULID.valid!("1234567890") }
+      expect_raises(IncorrectLength) { ULID.valid!("") }
+    end
+
+    it "should detect invalid characters" do
+      expect_raises(InvalidChars) { ULID.valid!("01!3EAF48P97R8MP9WS8MHDTZ3") }
+      expect_raises(InvalidChars) { ULID.valid!("01I3EAF48P97R8MP9WS8MHDTZ3") }
+      expect_raises(InvalidChars) { ULID.valid!("01O3EAF48P97R8MP9WS8MHDTZ3") }
+    end
+  end
+
+  describe ".seed_time" do
```

Comment:
Would be good to include a round trip test here (e.g. create a ULID that encodes a specific time, then decode this and ensure they match).